src/config_validator.py

```
@@ -0,0 +1,156 @@
+            float: Validated rate limit value
+        
+        Raises:
+            ConfigValidationError: If rate limit is invalid
+        """
+        try:
+            limit_float = float(rate_limit)
+            if limit_float <= 0 or limit_float > 100:  # Reasonable rate limit range
+                raise ConfigValidationError(f"Invalid rate limit value: {rate_limit}")
+            return limit_float
+        except (TypeError, ValueError):
+            raise ConfigValidationError(f"Invalid rate limit value: must be a numeric value, got {type(rate_limit)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)